src/components/table/table.tsx

```diff
@@ -37,18 +50,79 @@ export default function Table(props: TableData) {
   const headers: TableHeader[] =
     props.headers || dataKeys.map((key) => ({ dataKey: key }))
 
-  const dataOfCurrentPage = props.data.slice(
-    currentPage * rowsPerPage,
-    currentPage * rowsPerPage + rowsPerPage
-  )
+  function handleSortRequest(dataKey: string) {
+    setSortingCriteria((prevCriteria) => {
+      const currentOrder = prevCriteria[dataKey]
+      let newSortingCriteria = {}
 
-  const dataToDisplay = rowsPerPage > 0 ? dataOfCurrentPage : props.data
+      switch (currentOrder) {
+        case Order.Ascending:
+          newSortingCriteria = { [dataKey]: Order.Descending }
+          break
+        case Order.Descending:
+          newSortingCriteria = {}
+          break
+        default:
+          newSortingCriteria = { [dataKey]: Order.Ascending }
+          break
+      }
+
+      return newSortingCriteria
+    })
+  }
 
   function handleChangeRowsPerPage(event: React.ChangeEvent<HTMLInputElement>) {
     setRowsPerPage(+event.target.value)
     setCurrentPage(0)
   }
 
+  function sortData(array: Record<string, string | number>[]) {
+    const sortingKey = Object.keys(sortingCriteria)[0]
+    const sortingOrder = sortingCriteria[sortingKey]
+
+    if (!sortingKey || !sortingOrder) {
+      return array
+    }
+
+    return array.slice().sort((firstItem, secondItem) => {
+      const sortingKey = Object.keys(sortingCriteria)[0]
+      const sortingOrder = sortingCriteria[sortingKey]
+      if (firstItem[sortingKey] < secondItem[sortingKey]) {
+        return sortingOrder === Order.Descending ? 1 : -1
+      }
+      if (firstItem[sortingKey] > secondItem[sortingKey]) {
+        return sortingOrder === Order.Descending ? -1 : 1
+      }
+      return 0
+    })
+  }
+
+  const sortedData = sortData(props.data)
+
+  const dataOfCurrentPage = sortedData.slice(
+    currentPage * rowsPerPage,
+    currentPage * rowsPerPage + rowsPerPage
+  )
+
+  const dataToDisplay = rowsPerPage > 0 ? dataOfCurrentPage : sortedData
+
+  function getHeaderIcon(dataKey: string) {
+    if (!sortingCriteria[dataKey]) {
+      return SortIcon
+    }
```

Comment:
what about the cases when sort order is set ? also, I dont think we need to create a component SortIcon if its only used once

Reply:
`TableSortLabel` has default arrow for `asc` and `desc`. I only want to cover the case for unset sorting criteria. When nothing is passed, the default icon will be used. The reason why I made `SortIcon` as a component is because `TableSortLabel` only accepts `elementType` as input. I tried to pass the material icon span directly and got an error